store/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonReader.java

```diff
@@ -101,6 +111,59 @@ public static CarbonReaderBuilder builder(String tablePath, String tableName) {
     return reader.readSchema();
   }
 
+  /**
+   * Read carbonindex file and return the schema
+   *
+   * @param indexFilePath complete path including index file name
+   * @return null, if the index file is not present in the path.
+   * List<ColumnSchema> from the index file.
+   * @throws IOException
+   */
+  public static List<ColumnSchema> readSchemaInIndexFile(String indexFilePath) throws IOException {
```

Comment:
Create a `CarbonSchemaReader` and put `readSchemaInIndexFile`, `readUserSchema`, `readSchemaInDataFile` into it

Reply:
I raise a new PR for it:https://github.com/apache/carbondata/pull/2353